pkg/apis/policy/v1alpha1/propagation_types.go

```diff
@@ -255,6 +255,7 @@ type ReplicaSchedulingStrategy struct {
 	// "Divided" divides replicas into parts according to number of valid candidate member
 	// clusters, and exact replicas for each cluster are determined by ReplicaDivisionPreference.
 	// +kubebuilder:validation:Enum=Duplicated;Divided
+	// +kubebuilder:default=Divided
```

Comment:
Emmm🤔, I think this solution does not have an effect on #3013 because the empty field is `ReplicaDivisionPreference`.

Reply:
Yes, It looks good for this instruction.

But it can't fix #3013.

For the `ReplicaDivisionPreference`, maybe we can't just set a default value by kubebuilder instruction, because it has a precondition that is `when replicaSchedulingType  is Divided`.

How about deal `replicaSchedulingType  ` with the mutating webhook?

Reply:
+1. Meanwhile, I suggest we add our own default function of all API objects in `zz_generated.defaults.go` which generated by `code-generator` and fill up with the default value in mutating webhook.

Reply:
I'm interested in the defaults thing, is there an example?

Reply:
Take `corev1.Pod` as an example. AFAIK, the `code-generator` helps us generate [zz_generated.defaults.go](https://github.com/kubernetes/kubernetes/blob/a991dff74c9b95f5628780c9da5ef06885cae81d/pkg/apis/core/v1/zz_generated.defaults.go#L32-L32) and we only need to implement the customized default function in [defaults.go](https://github.com/kubernetes/kubernetes/blob/a991dff74c9b95f5628780c9da5ef06885cae81d/pkg/apis/core/v1/defaults.go#L31-L31). Once the scheme is added, we could use [these codes](https://github.com/kubernetes/kubernetes/blob/a991dff74c9b95f5628780c9da5ef06885cae81d/vendor/k8s.io/apimachinery/pkg/runtime/scheme.go#L344-L350) to dump all default values into an object.

Reply:
> the code-generator helps us generate [zz_generated.defaults.go](https://github.com/kubernetes/kubernetes/blob/a991dff74c9b95f5628780c9da5ef06885cae81d/pkg/apis/core/v1/zz_generated.defaults.go#L32-L32)

https://pkg.go.dev/k8s.io/code-generator/cmd/defaulter-gen

this?

Reply:
Right, I have used this tool before.